src/main/java/Parser.java

```diff
@@ -0,0 +1,63 @@
+import java.text.ParseException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Parser parses a string input and turn it into a Command.
+ */
+public class Parser {
+
+    /**
+     * Change date and time into LocalDateTime.
+     *
+     * @param dateAndTime String array consist of date as first element and time as second element
+     * @return LocalDateTime
+     * @throws NumberFormatException
+     */
+    public static LocalDateTime changeDateAndTime(String[] dateAndTime) throws NumberFormatException {
+        String date = dateAndTime[0];
+        String time = dateAndTime[1];
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+        LocalDateTime localDateAndTime = LocalDateTime.parse((date + " " + time), formatter);
+        return localDateAndTime;
+    }
+
+    /**
+     * Parse a string into Command.
+     * If the first letter is not recognised, throws InvalidFirstDukeException.
+     *
+     * @param fullCommand String
+     * @return Command
+     * @throws InvalidFirstDukeException
+     */
+    public static Command parse(String fullCommand) throws InvalidFirstDukeException {
+        String[] commandArr = fullCommand.split(" "); // split input into string array
+        Command command;
+
+        if (commandArr[0].equals("bye")) {
+            command = new Command(CommandType.EXITDUKE, null);
+        } else if (commandArr[0].equals("list")) {
+            command = new Command(CommandType.PRINTALLTASKS, null);
+        } else if (commandArr[0].equals("done")) {
+            command = new Command(CommandType.MARKASDONE, commandArr);
+        } else if (commandArr[0].equals("delete")) {
+            command = new Command(CommandType.DELETETASK, commandArr);
+        } else if (commandArr[0].equals("todo")) {
+            command = new Command(CommandType.ADDTODO, commandArr);
+        } else if (commandArr[0].equals("deadline")) {
+            command = new Command(CommandType.ADDDEADLINE, commandArr);
+        } else if (commandArr[0].equals("event")) {
+            command = new Command(CommandType.ADDEVENT, commandArr);
+        } else if (commandArr[0].equals("find")) {
+            command = new Command(CommandType.FINDTASK, commandArr);
+        } else {
+            throw new InvalidFirstDukeException();
+        }
+        return command;
```

Comment:
There is no coding violation here but would it better if you change the multiple else-if statements to a switch statement and handle each different case within the switch statement? This might help to improve code readability and make the code even cleaner.